src/common/string-builder.ts

```diff
@@ -81,8 +82,39 @@ class StringBuilder {
         return this.printQueue.join(' ');
     }
 
-   combineURLs(baseURL: string, relativeURL?: string): string {
-        return relativeURL ? new URL(relativeURL, baseURL || undefined).toString() : baseURL;
+    combineURLs(baseURL: string, relativeURL?: string): string {
+        const createURL = (url: string) => {
+            try {
+                return new URL(url)
+            } catch (e) {
+                return null
+            }
+        }
+
+        let url: string = '';
+        const isFirstUrl = (baseURL !== '' && baseURL !== undefined);
+        const firstURL = isFirstUrl ? new URL(baseURL) : null;
+        const isRelativeUrl = relativeURL && relativeURL !== '';
```

Comment:
`URL` and `url` are being mixed.

Reply:
and?

Reply:
your choice. I think URL is more consistency.